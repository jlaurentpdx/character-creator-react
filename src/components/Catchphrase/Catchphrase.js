export default function Catchphrase({
  label,
  inputText,
  buttonText,
  setInputText,
  setCatchphraseList,
}) {
  const clickHandler = () => {
    setCatchphraseList((prevState) => [...prevState, inputText]);
    setInputText('');
  };

  return (
    <label>
      {label}
      <input type="text" value={inputText} onChange={(e) => setInputText(e.target.value)} />
      <button onClick={clickHandler}>{buttonText}</button>
    </label>
  );
}
