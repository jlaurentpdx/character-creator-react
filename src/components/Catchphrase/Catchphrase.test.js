import { render, screen } from '@testing-library/react';
import Catchphrase from './Catchphrase';

test('Catchphrase should render a label, input, and button', () => {
  const { container } = render(
    <Catchphrase label="Add a catch phrase" inputText="in dog we trust" buttonText="Add" />
  );
  screen.getByText('Add a catch phrase');
  screen.getByDisplayValue('in dog we trust');
  screen.getByText('Add');

  expect(container).toMatchSnapshot();
});
