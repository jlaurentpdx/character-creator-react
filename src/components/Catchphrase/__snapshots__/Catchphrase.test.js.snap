// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catchphrase should render a label, input, and button 1`] = `
<div>
  <label>
    Add a catch phrase
    <input
      type="text"
      value="in dog we trust"
    />
    <button>
      Add
    </button>
  </label>
</div>
`;
